### PR TITLE
Resolve Issue IBM-Swift/Kitura#640

### PR DESCRIPTION
**Pull request by shmuelk**
Enable on Linux to use DispatchSource instead of epoll via a compile time switch

## Description
DispatchSource related code for keep alive is now selected if on MacOS and on Linux if the compile time switch GCD_ASYNCH is specified.

## Motivation and Context
This resolves IBM-Swift/Kitura#640

## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Sources/KituraNet/IncomingSocketManager.swift [hunk 1]
```diff
@@ -16,46 +16,60 @@
...
     
     /// Interval at which to check for idle sockets to close
-    let keepAliveIdleCheckingInterval: NSTimeInterval = 60.0
+    let keepAliveIdleCheckingInterval: TimeIntervalType = 60.0
     
     /// The last time we checked for an idle socket
-    var keepAliveIdleLastTimeChecked = NSDate()
+    var keepAliveIdleLastTimeChecked = DateType()
+    
+    #if GCD_ASYNCH
```
Comment (ianpartridge) [inline Sources/KituraNet/IncomingSocketManager.swift]: These two lines can just be `#if os(Linux)`

Comment (ianpartridge) [inline Sources/KituraNet/IncomingSocketManager.swift]: Ignore more, I'm wrong :)

Comment (ianpartridge) [inline Sources/KituraNet/IncomingSocketManager.swift]: `#if os(Linux) && !GCD_ASYNCH` might be clearer though.

File: Sources/KituraNet/HTTP/IncomingHTTPSocketHandler.swift [hunk 2]
```diff
@@ -135,13 +168,36 @@ class IncomingHTTPSocketHandler: IncomingSocketHandler {
...
+            source.cancel()
+        #elseif GCD_ASYNCH
+	    dispatch_source_cancel(source!)
+        #else
+            dispatchSourceCancelHandler()
+        #endif
+    }
+    
+    /// DispatchSource cancel handler
+    private func dispatchSourceCancelHandler() {
```
Comment (ianpartridge) [inline Sources/KituraNet/HTTP/IncomingHTTPSocketHandler.swift]: I think this should be `handleCancel()`, to match with `handleRead()`?

File: Sources/KituraNet/IncomingSocketManager.swift [hunk 3]
```diff
@@ -16,46 +16,60 @@
 
 
 #if os(Linux)
+    import Glibc
+    import CEpoll
```
Comment (ianpartridge) [inline Sources/KituraNet/IncomingSocketManager.swift]: We don't need this if GCD_ASYNCH is set.

File: Sources/KituraNet/IncomingSocketManager.swift (hunk 1 above)
Comment (shmuelk) [inline Sources/KituraNet/IncomingSocketManager.swift]: It would if you could.....

**Pull request merged**